src/object/srv_ec_aggregate.c

```diff
@@ -1354,24 +1502,28 @@ agg_peer_update(struct ec_agg_entry *entry, bool write_parity)
 	rc = pool_map_find_failed_tgts(agg_param->ap_pool_info.api_pool->sp_map,
 				       &targets, &failed_tgts_cnt);
 	if (rc) {
-		D_ERROR("pool_map_find_failed_tgts failed: "DF_RC"\n",
-			DP_RC(rc));
+		D_ERROR(DF_UOID" pool_map_find_failed_tgts failed: "DF_RC"\n",
+			DP_UOID(entry->ae_oid), DP_RC(rc));
 		goto out;
 	}
 	if (targets != NULL) {
-		int i;
-
-		for (i = 0; i < failed_tgts_cnt; i++) {
-			if (targets[i].ta_comp.co_rank ==
-			    entry->ae_peer_pshards[0].sd_rank) {
-				D_ERROR("peer parity tgt failed\n");
-				rc = -1;
-				goto out;
+		for (peer = 0; peer < p; peer++) {
+			peer_loc = &entry->ae_peer_pshards[peer];
+			for (i = 0; i < failed_tgts_cnt; i++) {
+				if (targets[i].ta_comp.co_rank ==
+				    peer_loc->sd_rank) {
+					D_ERROR(DF_UOID" peer parity tgt "
+						"failed rank %d, tgt_idx %d.\n",
+						DP_UOID(entry->ae_oid),
+						peer_loc->sd_rank,
+						peer_loc->sd_tgt_idx);
+					rc = -1;
```

Comment:
probably choose a different error code, like -1001 or DER_EIO?

Reply:
right, probably, I just use the same err code as original code, can change it later.